gix-blame/tests/fixtures/make_blame_repo.sh

```diff
@@ -6,18 +6,19 @@ git init -q
 git config merge.ff false
 
 git checkout -q -b main
-git commit -q --allow-empty -m c1
-git tag at-c1
-git commit -q --allow-empty -m c2
-git commit -q --allow-empty -m c3
-git commit -q --allow-empty -m c4
 
-git checkout -q -b branch1
-git commit -q --allow-empty -m b1c1
-git tag at-b1c1
-git commit -q --allow-empty -m b1c2
+echo "line 1" >> file.txt
+git add file.txt
+git commit -q -m c1
 
-git checkout -q main
-git commit -q --allow-empty -m c5
-git tag at-c5
-git merge branch1 -m m1b1
+echo "line 2" >> file.txt
```

Comment:
I'd do the same - start extremely simple, maybe do the first rough implementation so it passes this test, and then think of some tougher cases to throw at it, validating that they still come out right.

Maybe it's worth investing into a baseline test which parses the output of `git blame` to get the expected output, which then has to be matched by the algorithm. These are typically the second stage as they are more obscure, but make it easier to get correct expectations for a lot of possibly complex inputs.

Reply:
Thanks for the input, sounds like a great plan!